app/Events/Mship/Endorsement/TierEndorsementAdded.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Events\Mship\Endorsement;
+
+use App\Models\Mship\Account;
+use App\Models\Mship\Account\Endorsement;
+use Illuminate\Broadcasting\InteractsWithSockets;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+class TierEndorsementAdded
+{
+    use Dispatchable, InteractsWithSockets, SerializesModels;
+
+    /**
+     * Create a new event instance.
+     */
+    public function __construct(private Endorsement $endorsement, private Account $account)
```

Comment:
Why are we passing both the endorsement and the account? Why not just accept the endorsement access the account from the endorsement?

Could they also be public so that they can be accessed without the getters?

Reply:
The account is eager loaded in the event constructor on the observer. The separation was just a bit more explicit to stop an N + 1.